dartagnan/src/main/java/com/dat3m/dartagnan/utils/GitInfo.java

```diff
@@ -11,31 +11,28 @@
 
 public class GitInfo {
 
-    private static final Logger logger = LogManager.getLogger(Dartagnan.class);
+    private static final Logger logger = LogManager.getLogger(GitInfo.class);
 
     static Properties properties = new Properties();
 
-    public static void CreateGitInfo() throws IOException {
+    public static void initGitInfo() throws IOException {
         try (InputStream is = Dartagnan.class.getClassLoader()
                 .getResourceAsStream("git.properties")) {
             if (is == null) {
+                logger.warn("Failed to load git.properties");
                 return;
             }
             properties.load(is);
-            logger.info("Git branch: " + properties.getProperty("git.branch"));
-            logger.info("Git commit ID: " + properties.getProperty("git.commit.id"));
         }
     }
 
-    public static String getGitId() throws IOException {
-        try (InputStream is = Dartagnan.class.getClassLoader()
-                .getResourceAsStream("git.properties")) {
-            if (is == null) {
-                return "unknown";
-            }
-            properties.load(is);
-            return properties.getProperty("git.commit.id");
-        }
+    public static void logGitInfo() {
+        logger.info("Git branch: " + properties.getProperty("git.branch"));
+        logger.info("Git commit ID: " + properties.getProperty("git.commit.id"));
+    }
+
+    public static String getGitId() {
+        return properties.getProperty("git.commit.id");
```

Comment:
I think you should do a `getProperty(..., defaultValue)` (maybe "unknown" as default?) because the loading of properties may fail in which the normal `getProperty` would return NULL.